sql/src/main/java/io/crate/planner/node/dql/UnionPhase.java

```diff
@@ -83,16 +85,28 @@ public boolean isFetchRequired() {
 
     @Override
     public void addProjection(Projection projection) {
-        super.addProjection(projection);
         if (projection instanceof FetchProjection) {
             fetchRequired = true;
-            // Add a new symbol as first which acts as a placeholder for the
-            // upstream inputId that is prepended in the union phase
-            List<Symbol> outputsWithPrependedInputId = new ArrayList<>(projections.get(0).outputs());
-            outputsWithPrependedInputId.add(0, Literal.ZERO);
 
-            //TODO: replace intputs/outputs on all projections.
+            // shift indices of output and orderBy input columns right by 1
+            // (inputId byte value will be injected by the union phase as the first column)
+            for (Projection existingProjection : projections) {
+                InputColumn.shiftRight(existingProjection.outputs());
+
+                List<Symbol> outputs = new ArrayList<>(existingProjection.outputs());
+                outputs.add(0, new InputColumn(0, DataTypes.BYTE));
+                existingProjection.outputs(outputs);
+
+                if (existingProjection instanceof TopNProjection) {
```

Comment:
this seems a bit fragile to me - what if we add another projection as an alternative to topN which could also have an orderBy?
Though I don't have a better solution :(